src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourcelinkTests.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+public class SourcelinkTests : SmokeTests
+{
+    private static string SourcelinkRoot { get; } = Path.Combine(Directory.GetCurrentDirectory(), "sourcelink");
+
+    private string sourcelinkToolPath;
+    private ConcurrentBag<string> failedFiles = new ConcurrentBag<string>();
+
+    public SourcelinkTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+    /// <summary>
+    /// Verifies that all symbols have valid sourcelinks.
+    /// </summary>
+    [SkippableFact(Config.ExcludeSourcelinkEnv, skipOnTrueEnv: true)]
+    public void VerifySourcelinks()
+    {
+        if (Directory.Exists(SourcelinkRoot))
+        {
+            Directory.Delete(SourcelinkRoot, true);
+        }
+        Directory.CreateDirectory(SourcelinkRoot);
+
+        sourcelinkToolPath = GetSourcelinkToolPath();
```

Comment:
Having a test method set a class field seems like an anti-pattern. It would cause problems if there are multiple test methods that are run in parallel. Can this be done in the constructor? Otherwise, it should just be passed as a parameter where it's needed.

Reply:
Yeah, I had this in a constructor, but eventually moved it out - will update.

Reply:
Fixed with https://github.com/dotnet/installer/pull/17309/commits/9ff1e023cececde9d7f33c7e5d4d9e093347c249